src/main/java/org/opentripplanner/model/Stop.java

```diff
@@ -0,0 +1,209 @@
+/*
+ * Copyright (C) 2011 Brian Ferris <bdferris@onebusaway.org>
+ * Copyright (C) 2012 Google, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
```

Comment:
Could the Apache-licensed classes be pulled into a separate sub-module?  If these classes are pulled into the same module and package structure and start to get refactored/modified/etc the license is going to get really blurry, as the project is under LGPL.  I've always tried to keep a hard line between components that have different licenses.

Reply:
@barbeau If by separate sub-module you mean a Maven sub-module, we went through a lot of effort to remove all sub-modules from OTP and I would be against reintroducing that structure. The alternative of putting the code in a completely separate Maven module would also make the build process much more complex because these model objects are so tightly coupled to all parts of OTP.

The license situation does deserve careful consideration but at least we know there is not a deep conflict here. OTP is under LGPL 3.0, and this OBA code is under Apache 2.0. Apache 2.0 is compatible with LGPL 3.0 (http://www.gnu.org/licenses/license-list.en.html#apache2), which is to say Apache-licensed code can be included in an LGPL project. However, as you say it starts to get messy when this code is modified. 

@bdferris is there any chance you and Google would allow us to re-license these model objects under LGPL for direct inclusion in OTP?

Reply:
I can maybe take a look, but I'm guessing getting Google to sign-off on a license change is a non-trivial / painful process.